fluent-dom/src/localization.js

```diff
@@ -55,17 +37,55 @@ export default class Localization {
    */
   async formatWithFallback(keys, method) {
     const translations = [];
+    let hasMissing = true;
+    let debugInfo = null;
+
     for (let ctx of this.ctxs) {
       // This can operate on synchronous and asynchronous
       // contexts coming from the iterator.
       if (typeof ctx.then === "function") {
         ctx = await ctx;
       }
-      const errors = keysFromContext(method, ctx, keys, translations);
-      if (!errors) {
+      const missingIds = keysFromContext(method, ctx, keys, translations);
+      if (missingIds === null) {
+        hasMissing = false;
```

Comment:
As I stated in my previous review, I don't think `hasMissing` does what you'd like it to do. It will be `true` if all locales have missing translations but their sum is complete. In which case `Could not resolve all keys.` is incorrect.

I would remove `debugInfo` and `hasMissing` entirely from this patch. Would something like the following be enough?

```js
// Always return a set from keysFromContext, even if empty.
const missingIds = keysFromContext(method, ctx, keys, translations);

// Maybe factor out to a separate function?
if (typeof console !== "undefined") {
  for (const missingId of missingIds) {
    console.warn(
      `[Fluent] Missing translations in ${locale}: ${Array.from(ids).join(", ")}`);
  }
}
```
```

Reply:
I stand corrected: we're passing `translations` to `keysFromContext` so that subsequent iterations of the loop will only grow the list of retrieved translations. This doesn't change my suggestion however :)